_posts/en/newsletters/2022-12-07-newsletter.md

```diff
@@ -0,0 +1,238 @@
+      actual miner's algorithms; would it have been better to
+      unify these two implementations through refactoring?"
+  a2="We only need to operate on a cluster and not the entire mempool,
+      and don't need to apply any of the checks that `BlockAssembler`
+      does. It was also suggested to do this calculation without
+      holding the mempool lock. We'd also need to change the block
+      assembler to be tracking bump fees rather than building the
+      block template; the amount of refactoring necessary was
+      equivalent to rewriting."
+  a2link="https://bitcoincore.reviews/26152#94"
+
+  q3="Why does the MiniMiner require an entire cluster? Why can’t it
+      just use the union of each transaction’s ancestor sets?"
+  a3="Some of the ancestors may already have been paid for by some of
+      their other descendants; we don't have to pay with our transaction.
+      So we need to include these other descendants in our calculations."
+  a3link="https://bitcoincore.reviews/26152#129"
+
+  q4="If transaction X has a higher _ancestor feerate_ than independent
+      transaction Y, is it possible for a miner to prioritize Y
+      over X (that is, mine Y before X)?"
+  a4="Yes. If some of Y's low-feerate ancestors have _other_ decendants that
+      are high feerate, Y doesn't need to \"pay\" for those ancestors;
+      we can, in effect, \"remove\" those ancestors from Y's ancestor set,
+      which would then increase Y's _effective_ ancestor feerate."
+  a4link="https://bitcoincore.reviews/26152#169"
+
+  q5="Can `CalculateBumpFees()` overestimate, underestimate, both, or
+      neither? By how much?"
+  a5="It will overestimate if two outputs with overlapping ancestry are
+      chosen, since they're bumped independently (without taking the shared
+      ancestry into account). Another way is if an ancestor acquires a
+      high feerate child after we've calculated the bump fees. The
+      participants couldn't think of a way bump fees could be
+      underestimated, but the question was left open."
+  a5link="https://bitcoincore.reviews/26152#194"
+
+  q6="The `MiniMiner` is given a list of UTXOs (outpoints) that the wallet
+      might be interested in spending. Given an outpoint, what are its 4
+      possible states?"
+  a6="It could be a confirmed, or unconfirmed (in the mempool), or unconfirmed
+      but already being spent by an existing transaction in the mempool,
+      or it could be an outpoint that we've never heard of."
+  a6link="https://bitcoincore.reviews/26152-2#21"
+
+  q7="What approach is taken in the \"Bump unconfirmed parent txs to target
+      feerate\" commit?"
+  a7="This commit is the main behavior change of the PR.
+      We take the information we've learned about our UTXOs and change their
+      effective values to reflect the amount of fees that go towards bumping
+      their ancestry to the target feerate. Then we run coin selection as before."
```

Comment:
I think you want to be a bit more precise what "the information we've learned about our UTXOs" is. I can imagine being confused as there isn't as much of a linear story
```suggestion
      We use the `MiniMiner` to calculate the bump fees (the fees needed to bump
      their respective ancestries to the target feerate) of each UTXO and
      deduct those from their effective values. Then we run coin selection as before."
```